tests/foreman/maintain/test_service.py

```diff
@@ -158,7 +159,13 @@ def test_positive_service_enable_disable(sat_maintain):
     assert 'FAIL' not in result.stdout
     assert result.status == 0
     sat_maintain.power_control(state='reboot')
-    result = sat_maintain.cli.Service.status(options={'brief': True, 'only': 'foreman.service'})
+    result, _ = wait_for(
+        sat_maintain.cli.Service.status,
+        func_kwargs={'options': {'brief': True, 'only': 'foreman.service'}},
+        fail_condition=lambda res: "FAIL" in res.stdout,
```

Comment:
Since you explicitely specified a single `fail_condition` and did not use `handle_exception: True` - wouldn't this mean that the `wait_for` might straight up fail if it runs while the VM is not yet completely booted and e.g. SSH is not yet running?

Reply:
I guess.... I can add handle_exception